### ISSUE TITLE
Output with a comma

### ISSUE THREAD

**Issue by deanc** [matryer/xbar#604]
I have the following script:

```
#!/bin/bash
curl -s https://raw.githubusercontent.com/owid/covid-19-data/master/public/data/vaccinations/country_data/Finland.csv | tail -n1 | awk -F "\"*,\"*" '{print $4}' | xargs printf "%'.f\n"
```

Unfortunately even though when running it manually it outputs the number with commas, bitbar is appearing to ignore/strip the comma(s) leaving just the number.

e.g. if script outputs `12,345` bitbar will show `12345`

Why is this? Thanks!

**Comment by matryer** [comment]
Great question - I have no idea.

Can you help me run that script?

<img width="1039" alt="image" src="https://user-images.githubusercontent.com/101659/105976796-81892680-6088-11eb-9d59-f3818fb107b1.png">

**Comment by deanc** [comment]
> Great question - I have no idea.
> 
> Can you help me run that script?
> 
> <img alt="image" width="1039" src="https://user-images.githubusercontent.com/101659/105976796-81892680-6088-11eb-9d59-f3818fb107b1.png">

Hi, sorry, the output changed. Try this:

```
#!/bin/bash
POPULATION=5518000
VACCINATIONS=$(curl -s https://raw.githubusercontent.com/owid/covid-19-data/master/public/data/vaccinations/country_data/Finland.csv | tail -n1 | awk -F "\"*,\"*" '{print $5}')
PERCENTAGE=$(echo "($VACCINATIONS/$POPULATION*100)" | bc -l | xargs printf %.2f)
VACCINATIONS_COMMA=$(echo -e $VACCINATIONS | xargs printf "%'.f")
echo "💉 $VACCINATIONS_COMMA ($PERCENTAGE%)|size=12"

```

**Comment by matryer** [comment]
Confirmed - it strips the comma, and shouldn't.

<img width="168" alt="image" src="https://user-images.githubusercontent.com/101659/105992406-fb2b0f80-609c-11eb-9a52-3e15ffc37548.png">

**Comment by matryer** [comment]
Cool plugin btw :)

**Comment by deanc** [comment]
Thanks, hopefully we can find a fix :) My guess is comma is some special token, but who knows :)

**Comment by matopeto** [comment]
sometihng about different locale env, so printf formats different for terminal and for bitbar?

**Comment by deanc** [comment]
> sometihng about different locale env, so printf formats different for terminal and for bitbar?

This is a pretty good shout. It'd be strange that whatever evaluation is happening to put in the bitbar is not the same as system though.

**Comment by matopeto** [comment]
@deanc  you can verify this by  running `locale` command and compare output from terminal and bash

**Comment by deanc** [comment]
@matopeto If i right click on the bitbar and click "run in termal" it displays the commas. Presuming this is the same mechanism then it's not that. It appears i'm using `sh` in iterm2 and bash in the built in terminal.

**Comment by matryer** [comment]
I think this will be fixed in the rewrite, coming soon - see #607

**Comment by vivekrp** [comment]
Is this fixed @matryer? I facing similar issue, 

echo "${cost%%.*}" | awk '{printf ("%'\''d\n", $0)}' should give me an answer like 49,772 and it does give that answer when running from terminal but in bitbar/xbar it shows 49772.
